jobrunner/record_stats.py

```diff
@@ -54,11 +59,8 @@ def read_job_metrics(job_id, **metrics):
 
 
 def write_job_metrics(job_id, metrics):
-    if _conn is None:
-        ensure_metrics_db()
-
     raw_metrics = json.dumps(metrics)
-    _conn.execute(
+    get_connection(readonly=False).execute(
```

Comment:
Rather than passing `initial` to get connection and having it execute every time would another approach be to wrap this write in a `try/except` which catches the table not existing (probably have to do string matching on the error text) and then creates them and retries.

Reply:
That is an option. 

The reason I did it in get_connection is that is only place that knows if it is a new connection or not.  But I think that adding the readonly connections muddies those waters a bit.